docs/source/rfc-configuration-overhaul.md

```diff
@@ -0,0 +1,912 @@
+
+### Naming Issues
+
+The configuration parameter naming convention poses the following challenges.
+
+#### Overcomplicated ENV Variables
+
+ENV variables for all configuration parameters are named according to the internal module names. These technical
+designations might resonate with developers but can be confusing or alienating for users not deeply immersed in the
+project's inner mechanisms. The
+[POSIX specification](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap08.html#tag_08) actively
+encourages environment variables with succinct names. Given that a number of standard environment variable names exist,
+and their value can be redefined on a per-process basis, it is recommended to use generic environment variables as much
+as possible, before using specific names prefixed with `IROHA_`.
+
+#### Inconsistent Naming of ENV Variables
+
+The output from the `iroha --help` command displays a mix of naming conventions:
+
+```
+Iroha 2 is configured via environment variables:
+    IROHA2_CONFIG_PATH is the location of your `config.json` or `config.json5`
+    IROHA2_GENESIS_PATH is the location of your `genesis.json` or `genesis.json5`
+...
+    IROHA_TORII: Torii (gateway) endpoint configuration
+    IROHA_SUMERAGI: Sumeragi (emperor) consensus configuration
+    IROHA_KURA: Kura (storage). Configuration of block storage
+    IROHA_BLOCK_SYNC: Block synchronisation configuration
+...
+```
+
+The inconsistency between the `IROHA2` and `IROHA` prefixes can be confusing. Of the two, the `IROHA` prefix is
+preferred, as users are discouraged from using Iroha 1 in any new projects.
+
+#### Inconsistent Naming of Configuration Parameters
+
+Current configuration demonstrates some inconsistencies in parameter naming, leading to potential user confusion. To
+address this, some refinements are proposed:
+
+- The `torii` section uses terms like `p2p_addr`, `api_url`, and `telemetry_url`. These are actually addresses in the
+  form of `localhost:8080` and not full URLs, which would include components like protocol and path. To standardize,
+  these should all be changed to use `addr` (e.g., `addr_p2p`, `addr_api`, `addr_telemetry`).
+- Consider moving addresses to their respective domain scopes. For example, instead of `torii.telemetry_url`, we could
+  have `telemetry.addr`.
+- In the `logger` section, the term `max_log_level` could be simplified to `log_level` or just `level` to make it more
+  concise without losing meaning. Although users are more likely to set this parameter using an environment variable.
+- In the `wsv` section, the field `wasm_runtime_config` has a redundant `_config` suffix. Given these are all
+  _configuration_ parameters, such suffixes can be eliminated for clarity.
+
+Implementing a consistent naming convention like this will simplify the user's configuration process and reduce
+ambiguity.
+
+#### Redundant SCREAMING_CASE in JSON Configuration
+
```

Comment:
Maybe we need rule of using the SCREAMING_CASE ?
For example, use it just for parameters that doesn't have the preset value and must be filled in by user?

Reply:
This is a creative suggestion, but I am not convinced that it would simplify the configuration process more than it would complicate it. The ideal configuration format should be both consistent and simple, and stick to a single, straightforward naming scheme. Instead of relying on syntax to highlight important fields, it's better to clearly identify those in the documentation. After all, this approach is somewhat unconventional and diverges from the norms you might see in many other software projects.